cloud/pkg/devicecontroller/controller/downstream.go

```diff
@@ -625,12 +625,12 @@ func ifTwinPresent(twin v1alpha1.Twin, newTwins []v1alpha1.Twin) bool {
 }
 
 // addUpdatedTwins is function of add updated twins to send to edge
-func addUpdatedTwins(new []v1alpha1.Twin, twin map[string]*types.MsgTwin, version string) {
+func addUpdatedTwins(newTwin []v1alpha1.Twin, twin map[string]*types.MsgTwin, version string) {
 	opt := false
 	optional := &opt
-	for _, dtwin := range new {
+	for i, dtwin := range newTwin {
 		expected := &types.TwinValue{}
-		expected.Value = &dtwin.Desired.Value
+		expected.Value = &newTwin[i].Desired.Value
```

Comment:
use dtwin here ?

Reply:
That is the fix which resolve the problem. Please have a look on this comment: https://github.com/kubeedge/kubeedge/pull/667#issuecomment-500704525